owncloudApp/src/main/java/com/owncloud/android/ui/adapter/ExpandableUploadListAdapter.kt

```diff
@@ -0,0 +1,743 @@
+                        if (thumbnail == null) {
+                            thumbnail = ThumbnailsCacheManager.mDefaultImg
+                        }
+                        val asyncDrawable = ThumbnailsCacheManager.AsyncThumbnailDrawable(
+                            parentActivity.resources,
+                            thumbnail,
+                            task
+                        )
+                        fileIcon.setImageDrawable(asyncDrawable)
+                        task.execute(fakeFileToCheatThumbnailsCacheManagerInterface)
+                    }
+                }
+
+                if ("image/png" == upload.mimeType) {
+                    fileIcon.setBackgroundColor(
+                        ContextCompat.getColor(parentActivity, R.color.background_color)
+                    )
+                }
+
+            } else if (fakeFileToCheatThumbnailsCacheManagerInterface.isImage) {
+                val file = File(upload.localPath)
+                // Thumbnail in Cache?
+                var thumbnail = ThumbnailsCacheManager.getBitmapFromDiskCache(
+                    file.hashCode().toString()
+                )
+                if (thumbnail != null) {
+                    fileIcon.setImageBitmap(thumbnail)
+                } else {
+                    // generate new Thumbnail
+                    if (allowedToCreateNewThumbnail) {
+                        val task = ThumbnailsCacheManager.ThumbnailGenerationTask(fileIcon)
+                        thumbnail = ThumbnailsCacheManager.mDefaultImg
+                        val asyncDrawable = ThumbnailsCacheManager.AsyncThumbnailDrawable(
+                            parentActivity.resources,
+                            thumbnail,
+                            task
+                        )
+                        fileIcon.setImageDrawable(asyncDrawable)
+                        task.execute(file)
+                        Log_OC.v(TAG, "Executing task to generate a new thumbnail")
+                    }
+                }
+
+                if ("image/png".equals(upload.mimeType, ignoreCase = true)) {
+                    fileIcon.setBackgroundColor(
+                        ContextCompat.getColor(parentActivity, R.color.background_color)
+                    )
+                }
+            } else {
+                fileIcon.setImageResource(
+                    MimetypeIconUtil.getFileTypeIconId(
+                        upload.mimeType,
+                        fileName
+                    )
+                )
+            }
+        }
+
+        return view
+    }
+
+    /**
+     * Gets the status text to show to the user according to the status and last result of the
+     * the given upload.
+     *
+     * @param upload        Upload to describe.
+     * @return Text describing the status of the given upload.
+     */
+    private fun getStatusText(upload: OCUpload): String {
+
+        var status: String
+        when (upload.uploadStatus) {
```

Comment:
Do we need that status variable? Couldn't it be reformat to `return when() { ... }`